src/zenml/integrations/kubernetes/orchestrators/manifest_utils.py

```diff
@@ -108,7 +259,9 @@ def build_pod_manifest(
     env: Optional[Dict[str, str]] = None,
     mount_local_stores: bool = False,
     owner_references: Optional[List[k8s_client.V1OwnerReference]] = None,
-) -> k8s_client.V1Pod:
+    auto_generate_image_pull_secrets: bool = True,
```

Comment:
I don't think this is the right place to put this. The purpose of this function is to create a configurable pod manifest. In my opinion, you should be able to pass previously generated image pull secrets into this function, they should not be created as part of it.